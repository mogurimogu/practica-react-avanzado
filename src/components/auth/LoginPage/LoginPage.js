import React from 'react';
import { connect } from 'react-redux';
import { authLogin, uiResetError } from '../../../store/actions';
import { getUi } from '../../../store/selectors';

import LoginForm from './LoginForm';

function LoginPage({ onLogin, loading, error, onErrorClose }) {
  return (
    <div>
      <LoginForm onSubmit={onLogin} />
      {loading && <p>...login in nodepop</p>}
      {error && (
        <div onClick={onErrorClose} style={{ color: 'red' }}>
          {error.message}
        </div>
      )}
    </div>
  );
}

const mapStateToProps = getUi;
const mapDispatchToProps = { onLogin: authLogin, onErrorClose: uiResetError };

export default connect(mapStateToProps, mapDispatchToProps)(LoginPage);
