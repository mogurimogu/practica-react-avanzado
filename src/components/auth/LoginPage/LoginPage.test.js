import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import renderer from "react-test-renderer";
import LoginPage from "./LoginPage";

describe("LoginPage", () => {
  test("snapshot", () => {

    const store = {
        getState: () => ({auth: true}),
        subscribe: () => {},
        dispatch: () => {},
    }

    const tree = renderer.create(  
        <Provider store={store}>
            <BrowserRouter>
                <LoginPage/>
            </BrowserRouter>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
