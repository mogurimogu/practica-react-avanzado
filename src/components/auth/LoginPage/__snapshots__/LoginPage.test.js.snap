// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <input
      checked={false}
      name="remember"
      onChange={[Function]}
      type="checkbox"
    />
    <button
      disabled={true}
    >
      Login
    </button>
  </form>
</div>
`;
